test/ui-test/testSuites/suite_messaging/tst_ChatFlow/test.feature

```diff
@@ -1,169 +1,173 @@
+         And the user edits the message at index 0 and changes it to <edited>
+         # TODO: Check last message is an edited one, now just checking the last message is the expected one but could not be an edited one.
+         Then the chat message <edited> is displayed as an edited one
+         Examples:
+		 | message | edited		  |
+		 | Edit me | Edited by me |
+
+    @relyon-mailserver
+    Scenario Outline: The user can reply to another message
+         When the user replies to the message at index 0 with <reply>
+         Then the chat message <reply> is displayed as a reply
+         Examples:
+		 | reply		   |
+		 | This is a reply |
+
+# TODO: This scenario should be extracted to a different feature file bc it doesn't accomplish the background steps since it needs to change/specify the chat room
+#    Scenario Outline: The user joins a room and marks it as read
+#         When the user joins chat room test
+#		 And the user marks the channel <channel> as read
+#         # TODO find a way to validate that it worked
+#         Examples:
+#		 | channel |
+#		 | test    |
+
+	@mayfail
+	# TODO: It works standalone but when it runs as part of the sequence, the action of delete is not done always. The popup option does not appear.
+    Scenario Outline: The user can delete his/her own message
+         When the user sends a chat message <message>
+         And the user deletes the message at index 0
+         Then the last message displayed is not <message>
+         Examples:
+		 | message			   |
+		 | random chat message |
+
+    Scenario: The user can clear chat history
+        When the user sends a chat message Hi hi
+        And the user sends a chat message testing chat
+        And the user sends a chat message history
+        And the user clears chat history
         Then the chat is cleared
 
-	@mayfail @merge
-    Scenario: User can send a gif
-          When the user opens the chat section
-          And user joins chat room automation-test
-          Then The user is able to send a gif message
-          When the user opens app settings screen
-          And the user opens the messaging settings
-          Then tenor GIFs preview is enabled
-
-	@mayfail @merge
-    Scenario Outline: User can activate image unfurling
-          When the user opens the chat section
-          And user joins chat room automation-test
-          And the user sends the chat message <image_url>
+	@mayfail
+	# TODO: Verification of gif sent fails. Review it.
+    Scenario: The user can send a GIF
+        When the user sends a GIF message
+        Then the GIF message is displayed
+        When the user opens app settings screen
+        And the user opens the messaging settings
+        Then tenor GIFs preview is enabled
+
+    @mayfail
+	# TODO: It works standalone but when it runs as part of the sequence, the action of activates link preview doesn't work.
+    Scenario Outline: The user can activate image unfurling
```

Comment:
WDYT with this:

`````
Given the user sends a chat message <image_url>
And the image <image_url> is not unfurled in the chat
When the user opens app settings screen
And the user opens the messaging settings
And the user activates link preview
And the user activates image unfurling
And the user opens the chat section
Then the image <image_url> is unfurled in the chat
`````

For me, since we are validating the complete sequence of activating an image unfurling, the steps to achive it start from the navigation to activate the link preview.

Reply:
I understand your point but in this case I will apply my suggestion since the scenario description is `The user can activate image unfurling` so I see the sequence of activating, as well, as an action.